src/background/errors.js

```diff
@@ -12,6 +12,8 @@ export class BlockedSiteError extends ExtensionError {}
 
 export class AlreadyInjectedError extends ExtensionError {}
 
+export class RequestError extends Error {}
```

Comment:
This `RequestError` name unfortunately has the same problem that I raised previously - it suggests the error might be _any_ kind of request error, but you're only using it to specifically mean that the request was canceled. I think something like `RequestCanceledError` makes it clear that the exception was raised _because_ the request was canceled, rather than when attempting to cancel the request. As evidence for this I'd point to a number of APIs in other languages having [CancelledError](https://docs.python.org/3/library/asyncio-exceptions.html#asyncio.CancelledError), [CanceledException](https://developer.android.com/reference/android/app/PendingIntent.CanceledException) or similarly named errors for this purpose.

An alternative direction which I think would be semantically clear would be to use the same `RequestError` type for all kinds of error making the request but add a field (`reason`?) that allows distinguishing cancellation from other reasons (eg. network failure, unexpected response structure). You'd have to make a bunch of changes in other places to do this though.

Reply:
👍🏼